hawkbit-rest/hawkbit-mgmt-resource/src/test/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtContentTypeTest.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2020 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.mgmt.rest.resource;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+import org.eclipse.hawkbit.mgmt.rest.api.MgmtRestConstants;
+import org.eclipse.hawkbit.repository.model.DistributionSet;
+import org.eclipse.hawkbit.rest.util.JsonBuilder;
+import org.eclipse.hawkbit.rest.util.MockMvcResultPrinter;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.boot.autoconfigure.web.servlet.HttpEncodingAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.servlet.server.Encoding;
+import org.springframework.context.annotation.Import;
+import org.springframework.hateoas.MediaTypes;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.util.Arrays;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+/**
+ * With Spring Boot 2.2.x the default charset encoding became deprecated. In
+ * hawkBit we want to keep the old behavior for now and still return the charset
+ * in the response, which is achieved through enabling {@link Encoding} via
+ * properties.
+ */
+@SpringBootTest(properties = { "server.servlet.encoding.charset=UTF-8", "server.servlet.encoding.force=true" })
+@Import(HttpEncodingAutoConfiguration.class)
```

Comment:
Why do we need to import an auto config within a tests? Isn't `server.servlet.encoding.enabled=true` the default?

Reply:
I think this won't guarantee that the server actually uses the correct encoding if you manually add it to the test

Reply:
We had to implement a workaround for the tests here as the download Tests were failing when setting server.servlet.encoding.enabled=true for all the tests, due to a bug in MockHttpServletResponse, that will be fixed within 5.2.9. With this test, we want to ensure that charset=utf-8 is still returned in the response, because we can't test it within the other tests right now.